### PR TITLE
Story/12640

### PR DESCRIPTION
**Pull request by RHODOK**

### CODE REVIEW

File: docs/source/primitives.rst [hunk 1]
```diff
@@ -384,6 +384,21 @@ Schema
 
 .. jsonschema:: ../../src/core/schema/components/selection.json
 
+DateTime
+------
+
+A DateTime component, providing datetime selection functionality. Three types accepted in datetime_type,
+'datetime', 'date' and 'time' which represent their namesake datetime types respectively. In the case of
+datetime and date a min/max date validator ADDNAME can be provided which will block out dates which cannot be
```
Comment (PeterAlabaster) [inline docs/source/primitives.rst]: Is `ADDNAME` here intentional?

File: src/core/components.py [hunk 2]
```diff
@@ -165,6 +167,30 @@ def get_validators(self):
...
 
 
+class DateTime(Input):
+    __slots__ = [
+        "datetime_type",
+    ]
+
+    def get_datetime_type(self, datetime_type):
+        validate = self.validate_datetime(datetime_type)
+        if validate:
```
Comment (PeterAlabaster) [inline src/core/components.py:176]: Is the check here needed?
```python
self.validate_datetime(datetime_type)
return datetime_type
```
won't return if the InvalidArguments exception is raised, no? or is this more for verbosity / code clarity?

**Review by PeterAlabaster — APPROVED** [review]
Couple minor comments, otherwise LGTM!

File: docs/source/primitives.rst (hunk 1 above)
Comment (RHODOK) [reply docs/source/primitives.rst]: no! good catch, I needed to check the name on the frontend

File: src/core/components.py (hunk 2 above)
Comment (RHODOK) [reply src/core/components.py:176]: yes, specificity

Comment (RHODOK) [reply src/core/components.py:176]: can remove, but follows pattern implemented elsewhere (selection component)

**Review by PeterAlabaster — APPROVED** [review]
~~Approving again after force push~~
Oops wrong PR

**Review by ws-mps — APPROVED** [review]

**Pull request merged**